### PR TITLE
Minimal pull request to enable native loading and Makefile for local testing

### PR DESCRIPTION
**Pull request by hexmode**
It would be great to use the same Makefile in the future to run the tests on github, but that isn't in this pull request.

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Holy f that is quite the Makefile

**Comment by hexmode** [comment]
Makefiles are just task-oriented shell scripts.  There is a lot of extra stuff in there, so I should trim it to just run the tests.

**Comment by hexmode** [comment]
Ok, this failure looks similar to what I get here.  I guess if it compares well with the most recent failures against master, it should be merged?

**Comment by hexmode** [comment]
Just checked: It doesn't look the same as failures against master.  I'll see what I can do about those.

**Comment by gesinn-it-gea** [comment]
@hexmode I made a PR to Docker Hub so that now MW 1.37 is also available. Feel free to add it to the test matrix.

https://hub.docker.com/_/mediawiki

File: composer.json [hunk 1]
```diff
@@ -94,16 +94,16 @@
 		"process-timeout": 0
 	},
 	"scripts": {
-		"test": "php ../../tests/phpunit/phpunit.php -c phpunit.xml.dist",
-		"phpunit": "php ../../tests/phpunit/phpunit.php -c phpunit.xml.dist",
-		"phpdbg": "phpdbg -qrr ../../tests/phpunit/phpunit.php -c phpunit.xml.dist",
+		"test": "php ${MW_INSTALL_PATH}/tests/phpunit/phpunit.php -c phpunit.xml.dist",
```
Comment (hexmode) [inline composer.json:97]: @kghbln @JeroenDeDauw Is this ok?  I was having trouble with symlinks and this made it work.

File: src/Localizer/LocalMessageProvider.php [hunk 2]
```diff
@@ -16,38 +16,40 @@ class LocalMessageProvider implements MessageLocalizer {
...
 	 *
 	 * @param string $file
-	 * @param string|null $languageCode
+	 * @param ?string $languageCode
 	 */
 	public function __construct( string $file, ?string $languageCode = null ) {
 		$this->file = $file;
 		$this->languageCode = $languageCode;
-		$this->languageFileDir = $GLOBALS['wgMessagesDirs']['SemanticMediaWiki'];
+		$this->languageFileDir = !is_array( $GLOBALS['wgMessagesDirs']['SemanticMediaWiki'] )
```
Comment (hexmode) [inline src/Localizer/LocalMessageProvider.php:50]: The access to $wgMessageDirs is scattered throughout and needs to be fixed.

**Comment by hexmode** [comment]
Yay! Tests pass on 1.35!

**Comment by hexmode** [comment]
> @hexmode I made a PR to Docker Hub so that now MW 1.37 is also available. Feel free to add it to the test matrix.

done.

File: composer.json (hunk 1 above)
Comment (JeroenDeDauw) [reply composer.json:97]: Seems reasonable enough to me. Have not tried it myself. Nowadays I use make like seen here https://github.com/ProfessionalWiki/ExternalContent/blob/master/Makefile

**Pull request merged**